languages/python/TreeSitter/Python/AST.hs

```diff
@@ -20,4 +23,4 @@ $(do
   pwd             <- runIO getCurrentDirectory
   let invocationRelativePath = takeDirectory (pwd </> currentFilename) </> "../../vendor/tree-sitter-python/src/node-types.json"
   input <- runIO (eitherDecodeFileStrict' invocationRelativePath)
-  either fail (traverse datatypeForConstructors) input)
+  either fail (fmap (concat @[]) . traverse (datatypeForConstructors Grammar.tree_sitter_python)) input)
```

Comment:
We’re going to eventually have AST modules for each supported language, each of which need to do this whole flow. I think we should therefore encapsulate the whole thing into something like:

```haskell
astDeclarationsForLanguage :: Ptr Language -> FilePath -> Q [Dec]
```

which we’d call in this module (and other AST modules).